lib/asciidoc-reference-check.js

```diff
@@ -1,456 +1,369 @@
-                              }
-                            );
-                          }
-                        }
-                      } catch (err) {
-                        //console.log("ERROR READING FILE", err);
-                        allGood = false;
-                        atom.notifications.addError(
-                          "Cannot find the file: `" + fullFilePath + "`.",
-                          {
-                            dismissable: true
-                          }
-                        );
-                      }
-                      resolve();
-                    } else {
-                      //console.log(item);
-                      var fullFilePath = item;
-                      try {
-                        var data = fs.readFileSync(fullFilePath, "utf8");
-
-                        if (data) {
-                          //all good
-                          //do nothing
-                          //console.log("LINK FINE: " + currentLink[1]);
-                        } else {
-                          //console.log("----------------------ERROR-----------------------")
-                          //console.log("Cannot find anchor!" + currentLink[1] + " in file " + item);
-                          allGood = false;
-                          atom.notifications.addError(
-                            "Cannot find the anchor: `" +
-                              currentLink[1] +
-                              "` in **" +
-                              fullFilePath +
-                              "**.",
-                            {
-                              dismissable: true
-                            }
-                          );
-                        }
-                      } catch (err) {
-                        //console.log("ERROR READING FILE", err);
-                        allGood = false;
-                        atom.notifications.addError(
-                          "Cannot find the file: `" + fullFilePath + "`.",
-                          {
-                            dismissable: true
-                          }
-                        );
-                      }
-                      resolve();
-                    }
-                  });
-                });
-              }
+    //check all internal references
+    var notFoundInternal = difference(internalRefs, anchorArray);
+    if (internalRefs[0] == null) {
+      atom.notifications.addSuccess(
+        '`' + logMsgFilePath + '`: No internal reference found.'
+      );
+    } else if (notFoundInternal[0] == null) {
+        atom.notifications.addSuccess(
+          '`' + logMsgFilePath + '`: All internal references are `OK`.'
+        );
+    } else {
+      notFoundInternal.forEach(function (it) {
+        //console.log('Cannot find the anchor: ' + it + ' in current file.');
+        atom.notifications.addError(
+          '`' + logMsgFilePath + '`: Cannot find anchor: `' + it + '` in current file.',
```

Comment:
With the updated notification, which has filename in the message itself, I guess we can change this to
```
'`' + logMsgFilePath + '`: Cannot find anchor: `' + it + '` .',
```

Reply:
I agree with you. It will be changed in the next pull request